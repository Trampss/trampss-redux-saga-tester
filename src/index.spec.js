/* eslint-env jest */
import tester from './index'

describe('index', () => {
  function* saga(param1, param2) {
    yield param1
    const variable = yield param2
    yield variable
  }

  it('should inject effects middleware', () => {
    const mocks = {
      call: [() => ({ first: 'mock' })],
    }

    expect(
      tester(saga)('foo', { '@@redux-saga/IO': true, CALL: true })(mocks),
    ).toMatchSnapshot()
  })
})
