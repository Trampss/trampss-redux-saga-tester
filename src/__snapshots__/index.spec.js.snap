// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should inject effects middleware 1`] = `
Array [
  Object {
    "done": false,
    "value": "foo",
  },
  Object {
    "raw": Object {
      "done": false,
      "value": Object {
        "@@redux-saga/IO": true,
        "CALL": true,
      },
    },
    "value": Object {
      "first": "mock",
    },
  },
  Object {
    "done": false,
    "value": Object {
      "first": "mock",
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;
